src/sonic-yang-models/yang-models/sonic-mirror-session.yang

```diff
@@ -0,0 +1,168 @@
+module sonic-mirror-session {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-mirror-session";
+    prefix sms;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    description
+        "SONiC Mirror session yang model";
+
+    revision 2021-06-15 {
+        description
+            "Initial revision.";
+    }
+
+    typedef session_type {
+        type enumeration {
+            enum ERSPAN {
+                description "ERSPAN session";
+            }
+            enum SPAN {
+                description "SPAN session";
+            }
+        }
+    }
+
+    typedef session_direction {
+        type enumeration {
+            enum RX {
+                description "RX direction";
+            }
+            enum TX {
+                description "TX direction";
+            }
+            enum BOTH {
+                description "Both RX and TX direction";
+            }
+        }
+    }
+
+    container sonic-mirror-session {
+
+        container MIRROR_SESSION {
+
+            list MIRROR_SESSION_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32 {
+                            error-message "Invalid mirror session name";
+                            error-app-tag session-name-invalid;
+                        }
+                    }
+                    description
+                        "Mirror session name.";
+                }
+
+                leaf type {
+                    type session_type;
+                    default "ERSPAN";
+                    description
+                        "Mirror session type.";
+                }
+
+                leaf src_ip {
```

Comment:
Please add the must check for src_ip and dst_ip ERSPAN as discussed

Reply:
Change to ip-address and enforce it is only ipv4 so that in future when supported we can remove the check

Reply:
Added ERSPAN checks. Changed to ip-address and added must condition to have only IPv4 Address for now.